### PR TITLE
[refactor] Separate texture args from scalar arg declaration

### PR DESCRIPTION
**Pull request by ailzhang**
But texture arg declaration is still hacky as it abuses `is_array`.
That should be fixed in the followup PRs.

Related issue = #

### CODE REVIEW

File: cpp_examples/run_snode.cpp [hunk 1]
```diff
@@ -122,7 +122,8 @@ void run_snode() {
     }
 
     kernel_ext = std::make_unique<Kernel>(program, builder.extract_ir(), "ext");
-    kernel_ext->insert_arg(get_data_type<int>(), true);
+    kernel_ext->insert_arr_arg(get_data_type<int>(), /*total_dim=*/1, {n});
+    ;
```
Comment (PENGUINLIONG) [inline cpp_examples/run_snode.cpp:126]: ^

File: tests/cpp/ir/ndarray_kernel.cpp [hunk 2]
```diff
@@ -22,7 +22,8 @@ std::unique_ptr<Kernel> setup_kernel1(Program *prog) {
   }
   auto block = builder1.extract_ir();
   auto ker1 = std::make_unique<Kernel>(*prog, std::move(block), "ker1");
-  ker1->insert_arg(get_data_type<int>(), /*is_array=*/true);
+  ker1->insert_arr_arg(get_data_type<int>(), /*total_dim=*/1, {1});
+  ;
```
Comment (PENGUINLIONG) [inline tests/cpp/ir/ndarray_kernel.cpp:26]: ^

File: tests/cpp/ir/ndarray_kernel.cpp [hunk 3]
```diff
@@ -40,8 +41,9 @@ std::unique_ptr<Kernel> setup_kernel2(Program *prog) {
   }
   auto block2 = builder2.extract_ir();
   auto ker2 = std::make_unique<Kernel>(*prog, std::move(block2), "ker2");
-  ker2->insert_arg(get_data_type<int>(), /*is_array=*/true);
-  ker2->insert_arg(get_data_type<int>(), /*is_array=*/false);
+  ker2->insert_arr_arg(get_data_type<int>(), /*total_dim=*/1, {1});
+  ;
```
Comment (PENGUINLIONG) [inline tests/cpp/ir/ndarray_kernel.cpp:45]: ^

**Review by PENGUINLIONG — APPROVED** [review]
lgtm, what are the semicolons tho?

File: cpp_examples/run_snode.cpp (hunk 1 above)
Comment (ailzhang) [reply cpp_examples/run_snode.cpp:126]: oops, probably I screwed up my search and replace :P Thansk!

Comment (ailzhang) [inline cpp_examples/run_snode.cpp:126]:
```suggestion
```

File: tests/cpp/ir/ndarray_kernel.cpp (hunk 2 above)
Comment (ailzhang) [inline tests/cpp/ir/ndarray_kernel.cpp:26]:
```suggestion
```

File: tests/cpp/ir/ndarray_kernel.cpp (hunk 3 above)
Comment (ailzhang) [inline tests/cpp/ir/ndarray_kernel.cpp:45]:
```suggestion
```

**Pull request merged**